Trigger on Git merge (#5244)
dbt-labs/docs.getdbt.com: website/docs/docs/deploy/merge-jobs.md

By using CD in dbt Cloud, you can take advantage of deferral to build only the edited model and any downstream changes. With merge jobs, state will be updated almost instantly, always giving the most up-to-date state information in [dbt Explorer](/docs/collaborate/explore-projects).

## Prerequisites
- You have a dbt Cloud account. 
- You have access to the beta release for this functionality. 

## Set up job trigger on Git merge {#set-up-merge-jobs}

1. On your deployment environment page, click **Create Job** > **Merge Job**. 
1. Options in the **Job settings** section:
    - **Job name** &mdash; Specify the name for the merge job.
    - **Description** &mdash; Provide a descripion about the job. 
    - **Environment** &mdash; By default, it’s set to your production environment.
1. In the **Git trigger** section, the **Run on merge** option is enabled by default. Every time a PR merges (to the main branch) in your Git repo, this job will get triggered to run. 

> 1. In the **Execution settings** section, add the dbt commands that you want to invoke when this job runs. By default, it includes the `dbt build --select state:modified+` command. This informs dbt Cloud to build only new or changed models and their downstream dependents. For more [commands](/docs/deploy/job-commands), click **Add command**.

Reviewer: Probably good to call out here that if using `state:modified+` they will need to set `Compare changes against an environment (Deferral)` which is automatically set to the current enviroment (and this is right below this execution settings now rather than buried in advanced settings which was a mistake we discovered from the beta).

---
title: "Merge jobs in dbt Cloud"
sidebar_label: "Merge jobs"
description: "Learn how to trigger a dbt job run when a Git pull request merges."
---

# Merge jobs in dbt Cloud <Lifecycle status="beta" />

> To implement a continuous development (CD) workflow in dbt Cloud, you can set up merge jobs that triggers a dbt job to run when Git pull requests are merged into production. This creates a seamless development experience where changes made in code will automatically update production data. 

Reviewer 2: Suggest breaking up this sentence a tiny bit:
```suggestion
You can set up a merge job to implement a continuous development (CD) workflow in dbt Cloud. The merge job triggers a dbt job to run when someone merges Git pull requests into production. This creates a seamless development experience where changes made in code will automatically update production data. 
```

By using CD in dbt Cloud, you can take advantage of deferral to build only the edited model and any downstream changes. With merge jobs, state will be updated almost instantly, always giving the most up-to-date state information in [dbt Explorer](/docs/collaborate/explore-projects).

## Prerequisites
- You have a dbt Cloud account. 
- You have access to the beta release for this functionality. 

## Set up job trigger on Git merge {#set-up-merge-jobs}

1. On your deployment environment page, click **Create Job** > **Merge Job**. 
1. Options in the **Job settings** section:
    - **Job name** &mdash; Specify the name for the merge job.
    - **Description** &mdash; Provide a descripion about the job. 
    - **Environment** &mdash; By default, it’s set to the environment you created the job from.
1. In the **Git trigger** section, the **Run on merge** option is enabled by default. Every time a PR merges (to the main branch) in your Git repo, this job will get triggered to run. 

> 1. In the **Execution settings** section, add the dbt commands that you want to invoke when this job runs. By default, it includes the `dbt build --select state:modified+` command. This informs dbt Cloud to build only new or changed models and their downstream dependents. For more [commands](/docs/deploy/job-commands), click **Add command**.

Reviewer 2: @nghi-ly Is this [repeated below](https://github.com/dbt-labs/docs.getdbt.com/pull/5244/files#diff-f28b1583630dc788056299dd4237e262b2417c0fd769d89a272c57e4bb65a08bR27) for a reason?

Author: great catch! thank you!

Author: removed duplicate

> 1. In the **Execution settings** section, add the dbt commands that you want to invoke when this job runs. By default, it includes the `dbt build --select state:modified+` command. This informs dbt Cloud to build only new or changed models and their downstream dependents. For more [commands](/docs/deploy/job-commands), click **Add command**.

Author: ```suggestion
```
